### ISSUE TITLE
Unintentionally integration_google and google_synchronization installed in parallel, now neither works without the other

### ISSUE THREAD

**Issue by ernolf** [nextcloud/integration_google#142]
At first I wasn't even aware that google_synchronization is a fork, I didn't even realize that they were two different apps, since they are realy way to similar.
Until I noticed at some point that the app no longer worked at all and a different oauth redirect address had to be specified on the Google API page as the one already stored.
Well, when I install google_synchronization without having integration_google installed (installed only, does not have to be enabled), it doesn't work at all. Nothing shows up on the admin page SERVER/settings/admin/connected-accounts and when I install integration_google without google_synchronization installed, an internal server error on the admin page is thrown and these log echos:

```
  "url": "/settings/admin/connected-accounts",
  "message": "include(): Failed opening '/var/www/nextcloud/apps/integration_google/lib//Settings/Admin.php' for inclusion (include_path='/var/www/nextcloud/apps/news/vendor/pear/net_url2:/var/www/nerdparty/pear/archive_tar:/var/www/nextcloud/3rdparty/pear/console_getopt:/var/www/nextcloud/3rdparty/pear/pear-core-minimal/src:/var/www/nextcloud/3rdparty/pear/pear_exception:/var/www/nextcloud/appsr/www/nextcloud/lib/composer/composer/ClassLoader.php#582",
```
and:
```
  "url": "/settings/admin/connected-accounts",
  "message": "include(/var/www/nextcloud/apps/integration_google/lib//Settings/Admin.php): Failed to open stream: No such file or directory at /var/www/nextcloud/lib/composer/composer/ClassLoader.php",
```

How can I get rid of this mess?
I am grateful for every hint.

**Comment by marcelklehr** [comment]
> google integration and the synchronization app

This is not supported. Either install on or the other.

**Comment by ernolf** [comment]
I solved the issue for me. These are the steps to be taken:

- completely remove both apps, so that they are not present in apps folder
- create backup with database dump
- do upgrade

after the upgrade it is clean.
Now one can install one or the other, not both, not one first - deinstall - and then the other, because then is messes up again.
If you want to try one and then the other one, you have to do this steps:

- try one of both apps, test it, try it, look if it works
- revert your nextcloud-version with the backup to the state before the update
- do update again and try the other one.

I would certainly find out a simpler way to reset the namespaces but did not dig into it so deep, because its not important for me, since both apps are broken at the moment. They do nothing at all. They did stop working. (for me)

My solution: I subscribe to my Google Calendar with the '*secret address in iCal format*'. You find that address when you go in browser view of your google calendar to the three dots behind the calendar you want to subscribe --> 'Settings and sharing' 

Subscribe in Nextcloud Calendar to that link and give it a good name and its own color.
A subscription gives you a read-only access to that calendar.

Then you only have to set a reasonable refreshrate. The default refreshrate for subscribed calendars is about one week, so once in a week it get refreshed.

If you want to shorten that refreshrate, you have two options:

1. Shorten refreshrate for all subscribed calendars like described in the manual: [Calendar / CalDAV — Nextcloud latest Administration Manual latest documentation](https://docs.nextcloud.com/server/latest/admin_manual/groupware/calendar.html#refresh-rate)
That wil change the refreshrate for all subscribed calendars for all users on your instance.

But subscribed calendars with religious or legal holidays or moon phases can be upated once a month or even once a year while subscribed personal callendars like from Google or Mailcow would have to be synchronized up to every 15 minutes or sometimes even more often depending on the individual case.
There is a hidden posibility to set the refreshrate for each single subscribed calendar apart. Unfortunately it hasn’t made it into the user interface. There is a column for it in the database, but you have to set it manually in the SQL console. But to make things easy, I wrote a script, that gives you the means to set the refreshrate of your subscribed calendars to your choice. Every single calendar to its own refreshrate:

2. Shorten refreshrate on a per calender basis with the script *nc-subscribed-calendar*

This is how to install the script:
```
wget -O /usr/local/bin/nc-subscribed-calendar https://global-social.net/apps/raw/s/nc-subscribed-calendar
chmod +x /usr/local/bin/nc-subscribed-calendar
```
(maybe you have to adjust the path to your nextcloud installation, which is set to */var/www/nextcloud*)

Now simply run it:

`nc-subscribed-calendar`

The script will guide you through the steps.

Restriction: Only for MySQL/MariaDB but if someone need it and uses postgre, it should not be much of an effort for him to adapt the script 😉

**Comment by marcelklehr** [comment]
> both apps are broken at the moment. They do nothing at all. They did stop working. (for me)

Feel free to open a new issue for this

**Comment by ernolf** [comment]
> > google integration and the synchronization app
> 
> This is not supported. Either install on or the other.

Of course, it's not supported, but still people are running into the trap of having both apps installed. I see that every day. At least I noticed it, many don't even notice that they have installed two different apps that cancel each other out, so it doesn't help at all if you simply close the issue succinctly, then nobody who made the same mistake will find it that quickly.

At least I described how to get out of the misery, that's what I asked for (and what I was able to quickly solve myself)
